### DISCUSSION TITLE
[Share] Auto-detect existing tags in the title and abstract

### DISCUSSION THREAD

**Discussion by rockcor** [windingwind/zotero-actions-tags#268] [Action Scripts]
### Description

In case anybody else has the demand that add existing tags to untagged papers. 
* case-insensitive

### Event

None

### Operation

Script

### Data

```javascript
/**
 * A description of this script.
 * @author rockcor
 * @usage automatically adding tags
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/268
 */


const window = require("window")
const console = window.console

var rows = await Zotero.DB.queryAsync("SELECT name AS tagname FROM tags");
var lines = [];
for (let row of rows) {
	lines.push(row.tagname);
}

if (!item) {
    return "item is empty";
}


const title = item.getField("title");
const abstract = item.getField("abstractNote");
let n = 0;
for (let t of lines) {
	const regex = RegExp(t,'i');
    if (regex.test(title) || (abstract && regex.test(abstract))) {
        item.addTag(t);
        n+=1;
    }
}

if (n>0) {
    return `success match and add ${n} tags`;
} else {
    return `no match, please add new tags manually`;
}
```

### Anything else

Reference:
* https://forums.zotero.org/discussion/97395/is-there-any-way-to-get-a-list-of-current-tags
* https://github.com/windingwind/zotero-actions-tags/discussions/119

**Comment by ulahcherubim** [comment]
Thank you for this. Which lines I should modify to get keywords for other languages?

**Reply by rockcor** [reply]
`var rows = await Zotero.DB.queryAsync("SELECT name AS tagname FROM tags");` 
Try to use more complicated SQL query. Hope the `language` column exists in `tags` table. If not, it totally depends on your language. In chinese, it's 

`select * from 表名 where 列名 like '%[吖-座]%' `

**Comment by antelamin** [comment]
is there a way to get this to ignore underscores in tags?

**Comment by belaaugustwalker** [comment]
How would I list words to ignore? Or restrict recognition to a whole word? As an example, I have the tag "trans" which I would like to apply to a title like "trans man," but not every article titled "transformation" or "transitive".

**Reply by rockcor** [reply]
You may try "trans<space>".
